pxfish/operation_type.py

```diff
@@ -33,130 +37,153 @@ def is_operation_type(path):
+
+def pull(*, session, path, category, name):
+    """Retrieves operation type, and calls function to write the associated files"""
+    retrieved_operation_type = get_operation_type(
+        session=session,
+        category=category, name=name)
+    write_files(session=session, path=path,
+                operation_type=retrieved_operation_type)
+
+
+def write_files(*, session, path, operation_type):
     """
     Writes the files associated with the operation_type to the path.
 
     Arguments:
-      path (string): the path to where the files will be written
-      operation_type (OperationType): the operation type being written
+        session (Session Object): Aquarium Session object
+        path (String): the path to where the files will be written
+        operation_type (OperationType): the operation type being written
     """
-    logging.info("writing operation type {}".format(operation_type.name))
+    logging.info('writing operation type %s', operation_type.name)
 
     category_path = create_named_path(path, operation_type.category)
     makedirectory(category_path)
 
-    path = create_operation_path(category_path, operation_type.name)
-    makedirectory(path)
+    path = create_named_path(
+        category_path, operation_type.name, subdirectory='operation_types')
 
+    makedirectory(path)
     code_names = operation_type_code_names()
 
     for name in code_names:
         code_object = operation_type.code(name)
         if not code_object:
             logging.warning(
-                "Missing {} code for operation type {}".format(
-                    operation_type.name, name)
-            )
-            continue
+                'Missing %s code for operation type %s -- creating file', operation_type.name, name)
+            create_code_object(
+                session=session,
+                name=name,
+                operation_type=operation_type
+                )
+            code_object = operation_type.code(name)
 
         file_name = "{}.rb".format(name)
+
         try:
-            code.write(path, file_name, code_object)
+            code.write(path=path, file_name=file_name, code_object=code_object)
         except OSError as error:
             logging.warning(
-                "Error {} writing file {} for operation type {}".format(
-                    error, file_name, operation_type.name))
+                'Error %s writing file %s for operation type %s',
+                error, file_name, operation_type.name)
             continue
         except UnicodeError as error:
-            message = "Encoding error {} writing file {} for operation type {}"
             logging.warning(
-                message.format(
-                    error, file_name, operation_type.name))
+                'Encoding error %s writing file %s for operation type %s',
+                error, file_name, operation_type.name)
             continue
 
+    sample_path = create_named_path(
```

Comment:
Let's not put these here. 

Look at pulling these from an operation type in the context where the operation type is pulled, and then writing to a `types` directory.
So, would write the types needed for the operation type (but not in the same directory as the operation type)